Optimize GH action

yetanotherparrot opened the pull request.

yetanotherparrot committed:
Renamed rust.yml workflow to "Code Quality Checks"

yetanotherparrot committed:
Rename GH workflow file

SpectralPixel commented:
Once again, too many PRs. Well, the workflows should be in `main`, so I guess there's not much that can be done...

Except for thinking through my PRs before submitting them of course. Too bad I find good ways to improve my stuff after handing it in...

yetanotherparrot committed:
Rename GH action job from "build" to "Enforce_Code_Quality"

yetanotherparrot committed:
test: Exclude `argentum_game` from testing in GH action
I hope this also skips the compilation of this crate, since most of the compile time is spent on it (and it doesn't even have tests!).

yetanotherparrot committed:
Don't include dependencies when running `cargo clippy`
I'm aware that this flag is implied, but once again I want to test if I can reduce compile times by explicity passing it.

yetanotherparrot committed:
Revert "Don't include dependencies when running `cargo clippy`"

This change did not reduce compile times. My understanding has been solidified.

This reverts commit 54b20b3e601b11c24f60ebeec6ad65ba151f302e.

yetanotherparrot committed:
Skip installing the Rust Toolchain in `Enforce_Code_Quality`
Doesn't this come preinstalled?

SpectralPixel commented:
I was able to shave off a couple minutes of compile times! :)
